### PR TITLE
Naga update, remove spirv_headers dependency

### PR DESCRIPTION
**Pull request by kvark**
**Connections**
Many changes went into Naga, including https://github.com/gfx-rs/naga/pull/81

**Description**
We'll get better SPIR-V parsing and even a bit of WGSL validation.

**Testing**
Working on it...

### CODE REVIEW

File: wgpu-core/Cargo.toml [hunk 1]
```diff
@@ -35,15 +35,15 @@ raw-window-handle = { version = "0.3", optional = true }
...
 thread-id = { version = "3", optional = true }
 tracing = { version = "0.1", default-features = false, features = ["std"] }
 tracing-subscriber =  { version = "0.2", optional = true }
 thiserror = "1"
 
 [dependencies.naga]
 git = "https://github.com/gfx-rs/naga"
-rev = "a9228d2aed38c71388489a95817238ff98198fa3"
+rev = "94802078c3bc5d138f497419ea3e7a869f10916d"
+features = ["spirv"]
```
Comment (cwfitzgerald) [inline wgpu-core/Cargo.toml:46]: I presume the plan going forward is to gate naga frontends/backends behind features?

Comment (kvark) [reply wgpu-core/Cargo.toml:46]: yes

**Comment by kvark** [comment]
Tested on wgpu-rs examples on top of bovine interverntion
bors r=cwfitzgerald

**Comment by kvark** [comment]
bors r=cwfitzgerald

**Pull request merged**